components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/SecureFamilyTools.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.studies.family.internal;
+
+import org.phenotips.data.Patient;
+import org.phenotips.studies.family.Family;
+import org.phenotips.studies.family.FamilyTools;
+import org.phenotips.studies.family.Pedigree;
+import org.phenotips.studies.family.exceptions.PTException;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.security.authorization.Right;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+/**
+ * A secure wrapper around FamilyTools which exposes a version of a Family object with
+ * no access to underlying XWikiDocument (implemented by {@link SecureFamily}).
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Component(roles = FamilyTools.class)
+@Named("secure")
+@Singleton
+public class SecureFamilyTools implements FamilyTools
```

Comment:
Agree, just Family classes (e.g. `FamilyTools`) were originally designed in a different way (compared to Patient classes\PatientRepository), and I noticed this discrepancy and did not like it, just as you said that was too many changes for one PR